Makefile

```diff
@@ -0,0 +1,61 @@
+VERSION := 0.19.1
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# ABI versioning
+SONAME_MAJOR := 0
+SONAME_MINOR := 0
+
+OBJ := build/Release/obj.target/tree_sitter_go_binding/src/*.o
+
+# OS-specific bits
+ifeq ($(shell uname),Darwin)
+	SOEXT = dylib
+	SOEXTVER_MAJOR = $(SONAME_MAJOR).dylib
+	SOEXTVER = $(SONAME_MAJOR).$(SONAME_MINOR).dylib
+	LINKSHARED += -dynamiclib -Wl,-install_name,$(LIBDIR)/libtree-sitter-go.$(SONAME_MAJOR).dylib
+else
+	SOEXT = so
+	SOEXTVER_MAJOR = so.$(SONAME_MAJOR)
+	SOEXTVER = so.$(SONAME_MAJOR).$(SONAME_MINOR)
+	LINKSHARED += -shared -Wl,-soname,libtree-sitter-go.so.$(SONAME_MAJOR)
+endif
+ifneq (,$(filter $(shell uname),FreeBSD NetBSD DragonFly))
+	PCLIBDIR := $(PREFIX)/libdata/pkgconfig
+endif
+
+%.o:
+	npm install
+
+all: libtree-sitter-go.a libtree-sitter-go.$(SOEXTVER)
+
+libtree-sitter-go.a: $(OBJ)
+	$(AR) rcs $@ $^
+
+libtree-sitter-go.$(SOEXTVER): $(OBJ)
+	$(CC) $(LDFLAGS) $(LINKSHARED) $^ $(LDLIBS) -o $@
+	ln -sf $@ libtree-sitter-go.$(SOEXT)
+	ln -sf $@ libtree-sitter-go.$(SOEXTVER_MAJOR)
+
+install: all
+	install -d '$(DESTDIR)$(LIBDIR)'
+	install -m755 libtree-sitter-go.a '$(DESTDIR)$(LIBDIR)'/libtree-sitter-go.a
+	install -m755 libtree-sitter-go.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter-go.$(SOEXTVER)
+	ln -sf libtree-sitter-go.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter-go.$(SOEXTVER_MAJOR)
+	ln -sf libtree-sitter-go.$(SOEXTVER) '$(DESTDIR)$(LIBDIR)'/libtree-sitter-go.$(SOEXT)
+	install -d '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter
+	install -m644 bindings/c/go.h '$(DESTDIR)$(INCLUDEDIR)'/tree_sitter/
+	install -d '$(DESTDIR)$(PCLIBDIR)'
+	sed -e 's|@LIBDIR@|$(LIBDIR)|;s|@INCLUDEDIR@|$(INCLUDEDIR)|;s|@VERSION@|$(VERSION)|' \
+	    -e 's|=$(PREFIX)|=$${prefix}|' \
+	    -e 's|@PREFIX@|$(PREFIX)|' \
+	    bindings/c/tree-sitter-go.pc.in > '$(DESTDIR)$(PCLIBDIR)'/tree-sitter-go.pc
```

Comment:
Just curious, could you explain more about the use case for this `install` action?

I'm not clear on when you'd want to actually install something like `tree-sitter-go` into a  system library folder, and what the conventions are for where it should be installed. Admittedly, I was never clear on when you'd use the library's `install` action either - that was added by some folks who maintain OS package management systems.

Does swift rely on the `.pc` file somehow? How is the file consumed in that case?

Reply:
I can only speak for myself, but I usually find it easier to build against binary dependencies when the libraries and headers are in a known location. That way, you only need one additional header and lib search path.

Swift can use any library accessible from C *provided* it has a clang module available. Strangely, the Swift Package Manager (SPM) cannot use modules for building/linking, but *can* use `.pc` files. I suspect this is a bug, and will eventually be resolved, as clang modules provide a large superset of what `.pc` files do. But, for now, the `.pc` is really handy for SPM support.

However, all that said, I do not know of a standardized way of installing a modulemap file into a system in a way that clang (or swift) will automatically find. That would be incredibly handy, though.